front_end/src/app/(campaigns-registration)/bridgewater/components/registration-status.tsx

```diff
@@ -0,0 +1,260 @@
+"use client";
+
+import React, { FC, PropsWithChildren, useState } from "react";
+import { BWRegisterButton, ChoicesButtons } from "./hero-section";
+import cn from "@/utils/cn";
+import { CurrentUser } from "@/types/users";
+import BaseModal from "@/components/base_modal";
+import {
+  RegistrationAndSignupForm,
+  RegistrationForm,
+} from "./registration-forms";
+import { CAMPAIGN_KEY } from "../constants";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faCheckCircle } from "@fortawesome/free-solid-svg-icons";
+import { faXmarkCircle } from "@fortawesome/free-regular-svg-icons";
+import { useRouter } from "next/navigation";
+import { useModal } from "@/contexts/modal_context";
+import Link from "next/link";
+
+interface RegisterAndStatusProps {
+  className?: string;
+  currentUser: CurrentUser | null;
+}
+
+export const RegisterAndStatus: FC<RegisterAndStatusProps> = ({
+  className,
+  currentUser,
+}) => {
+  const [dialogState, setDialogState] = useState<{
+    visible: boolean;
+    content: React.ReactNode;
```

Comment:
Is there a particular reason why we’re storing JSX as part of the React state? It seems unintuitive and might create issues, especially if we need to trigger effects based on `dialogState`.

What are your thoughts on this refactor: instead of storing content as JSX, we could use a `type` field (string or enum) and map the possible types to the corresponding modal content using an object (or `Map`)?

Reply:
Nog, good point. Good idea with the refactor, I will create actually two dialogs to make it more clear, each with their own state.